infra/aws/terraform/artifacts.k8s.io/s3/bucket.tf

```diff
@@ -0,0 +1,129 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+resource "aws_s3_bucket" "artifacts-k8s-io" {
+  provider = aws
+  bucket = "${var.prefix}artifacts-k8s-io-${data.aws_region.current.name}"
+}
+
+resource "aws_s3_bucket_acl" "artifacts-k8s-io" {
+  provider = aws
+  bucket = aws_s3_bucket.artifacts-k8s-io.bucket
+  # This clears the ACL list, so we can apply object_ownership = "BucketOwnerEnforced"
```

Comment:
I'm not sure that this does have that effect.

Reply:
Well it seemed to work when I accidentally applied an ACL the first time :-)

I hit the InvalidBucketAclWithObjectOwnership described [here](https://docs.aws.amazon.com/AmazonS3/latest/userguide/object-ownership-error-responses.html)

And took the hint from [here](https://github.com/hashicorp/terraform-provider-aws/issues/26164#issuecomment-1260995693)

If there's a better way, that would be great!